olive/passes/pytorch/lora.py

```diff
@@ -187,26 +180,115 @@ def _default_config(accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigPa
+        """Collate a batch of samples into a padded batch of tensors.
+
+        Add padding to the input_ids, attention_mask and labels.
+        """
+        from torch.nn.utils.rnn import pad_sequence
+
+        input_ids = [sample["input_ids"] for sample in batch]
+        attention_mask = None
+        if "attention_mask" in batch[0]:
+            attention_mask = [sample["attention_mask"] for sample in batch]
+        label_col = "labels" if "labels" in batch[0] else "label"
+        if label_col not in batch[0]:
+            raise ValueError("Batch does not contain 'labels' or 'label' column.")
+        labels = [sample[label_col] for sample in batch]
+
+        # apply padding and add to batch
+        new_batch = {
+            "input_ids": pad_sequence(input_ids, batch_first=True, padding_value=tokenizer.pad_token_id),
+            "labels": pad_sequence(labels, batch_first=True, padding_value=IGNORE_INDEX),
+        }
+        if attention_mask:
+            new_batch["attention_mask"] = pad_sequence(attention_mask, batch_first=True, padding_value=0)
+        return new_batch
+
+    @staticmethod
+    def get_datasets(config: ConfigBase, data_root: str) -> tuple:
+        """Load training and evaluation datasets."""
+        train_data_config = config.train_data_config
+        eval_data_config = config.eval_data_config
+        eval_dataset_size = config.eval_dataset_size
+
+        # load training dataset
+        train_data_container = train_data_config.to_data_container()
+        train_dataset = train_data_container.pre_process(train_data_container.load_dataset(data_root))
+        train_dataset = train_dataset.to_hf_dataset(label_name="labels")
+
+        # load evaluation dataset if needed
+        if eval_data_config:
+            # eval data config has been provided
+            eval_data_container = eval_data_config.to_data_container()
+            eval_dataset = eval_data_container.pre_process(eval_data_container.load_dataset(data_root))
+            eval_dataset = eval_dataset.to_hf_dataset(label_name="labels")
+        elif eval_dataset_size:
+            if eval_dataset_size >= 1:
+                # when eval_dataset_size is an integer, it is the number of samples
+                eval_dataset_size = int(eval_dataset_size)
+            # eval data config has not been provided, but eval_dataset_size has been provided
+            split_data = train_dataset.train_test_split(
+                test_size=eval_dataset_size, shuffle=True, seed=config.training_args.data_seed
+            )
+            train_dataset = split_data["train"]
+            eval_dataset = split_data["test"]
+        else:
+            # eval data config has not been provided, and eval_dataset_size has not been provided
+            eval_dataset = None
+
+        return train_dataset, eval_dataset
+
+    def enable_lora(
+        self, model, tokenizer, task: str, config: ConfigBase, target_modules: Optional[List[str]]
     ) -> PyTorchModel:
```

Comment:
the output of this method is not `PytorchModel` but `PeftModel` or simply `torch.nn.Module`

Reply:
could you also add type hints for `model` and `tokenizer` so that we don't confuse the type for `model`? `run_for_config` has it as pytorchmodel but now it's a torch model or transformers model.

Reply:
Will do